### PR TITLE
Make airship control plane node count configurable

### PR DESCRIPTION
**Pull request by jesusaurus**
Add an ansible variable for the number of caasp workers to use as
airship control plane workers instead of hard-coding to 2.

### CODE REVIEW

**Comment by jsuchome** [comment]
1. Could you also adapt playbooks/roles/caasp4/templates/inventory-caasp4.yml.j2 ?
2. What if we are not deploying on openstack?

**Comment by jesusaurus** [comment]
@jsuchome
1. done
2. I'm not very familiar with that use case, I thought inventory files were manually created in that case. I've renamed the variable and moved it to common-vars

File: vars/common-vars.yml [hunk 1]
```diff
@@ -125,3 +125,5 @@ scale_profile: "{{ lookup('env','scale_profile') | default('minimal', true) }}"
 
 #Set timeout for helm cleanup.
 helm_delete_timeout: 900
+
+airship_ucp_workers: 2
```
Comment (jsuchome) [inline vars/common-vars.yml]: The meaning is actually "control workers", right? As opposed to "compute workers", if I get it right. So it is a subset of "caasp workers" and current variable name does not make it clear IMHO

File: playbooks/openstack-caasp_instance.yml [hunk 2]
```diff
@@ -116,8 +116,8 @@
               {% endfor %}
 
               airship-openstack-control-workers:
-                hosts: &firsttwoworkers
-              {% for node in workers %}{% if loop.index <= 2 %}
+                hosts: &first_workers
```
Comment (jsuchome) [inline playbooks/openstack-caasp_instance.yml]: What is actually the meaning of this? I do not understand the syntax...

Comment (jesusaurus) [reply playbooks/openstack-caasp_instance.yml]: This is a yaml "anchor" 
https://confluence.atlassian.com/bitbucket/yaml-anchors-960154027.html

Comment (jsuchome) [reply playbooks/openstack-caasp_instance.yml]: Thanks!

File: vars/common-vars.yml [hunk 3]
```diff
@@ -143,6 +143,10 @@ scale_profile: "{{ lookup('env','scale_profile') | default('minimal', true) }}"
 #Set timeout for helm cleanup.
 helm_delete_timeout: 900
 
+# Number of caasp workers to use for the control plane,
+# the remainder will be used as computer hosts
+airship_ucp_control_workers: 2
```
Comment (toabctl) [inline vars/common-vars.yml]: Why common-vars? This is airship specific...

File: playbooks/roles/caasp4/templates/inventory-caasp4.yml.j2 [hunk 4]
```diff
@@ -32,15 +32,15 @@ airship-openstack-control-workers:
...
+  hosts: *first_workers
   vars:
     ansible_user: sles
     ansible_python_interpreter: /usr/bin/python3
 
 airship-openstack-compute-workers:
   hosts:
 {% for worker_ip in _terraform_json_output.ip_workers.value %}
-{% if loop.index0 > 1 %}
+{% if loop.index0 >= airship_ucp_control_workers %}
```
Comment (toabctl) [inline playbooks/roles/caasp4/templates/inventory-caasp4.yml.j2]: urgs. so we (again) include $random variables from everywhere in specific roles. But not your fault..

File: vars/common-vars.yml (hunk 3 above)
Comment (jesusaurus) [reply vars/common-vars.yml]: I put it under the airship-only section of common-vars https://github.com/SUSE-Cloud/socok8s/blob/master/vars/common-vars.yml#L56; we can move that entire section to a new file, but I believe that is a separate task from this PR.

**Pull request merged**